go/arrow/internal/arrjson/arrjson_test.go

```diff
@@ -5085,3 +5086,352 @@ func makeExtensionsWantJSONs() string {
   ]
 }`
 }
+
+func makeUnionWantJSONs() string {
```

Comment:
Perhaps make some of the child entries null? VALIDITY seems to be always 1 below.

Reply:
The top level validity of a union array should always be all 1's (since the top level validity bitmap is always left out for a union array). But i can definitely make some of the children entries null.

Reply:
updated with child nulls